react/features/welcome/components/native/settings/components/SettingsView.js

```diff
@@ -595,4 +630,21 @@ function _mapStateToProps(state) {
     };
 }
 
-export default translate(connect(_mapStateToProps)(withTheme(SettingsView)));
+export default translate(connect(_mapStateToProps)(withTheme(props => {
+    const isFocused = useIsFocused();
+    const { isTabNavigatorScreen, onSettingsScreenFocused } = props;
+
+    useEffect(() => {
+        if (isTabNavigatorScreen) {
+            if (isFocused) {
+                onSettingsScreenFocused(true);
+            } else {
+                onSettingsScreenFocused(false);
+            }
+        }
+    });
```

Comment:
Shouldn't we filter by isTabNavigatorScreen and isFocused ?

Reply:
We can do that yes.

Reply:
I think it better if we keep this way because the most we have to check first if it's a tab screen or a navigator screen. 
Afterwards we apply the focus logic.